libr/socket/socket.c

```diff
@@ -469,8 +493,14 @@ R_API bool r_socket_listen(RSocket *s, const char *port, const char *certfile) {
 		return false;
 	}
 #endif
-	if ((s->fd = socket (AF_INET, SOCK_STREAM, IPPROTO_TCP)) == R_INVALID_SOCKET) {
-		return false;
+	if (s->proto == R_SOCKET_PROTO_UDP) {
```

Comment:
better use switch(s->proto) { because the "else" of UDP is not TCP, there's also UNIX sockets, and other kind of sockets may be supported in the future

Reply:
also may be better to write a function that converts an RSocketProto into posix IPPROTO_

Reply:
R_SOCKET_PROTO_* is defined with IPPROTO_* in r_socket.h already. I will just use that instead.
https://github.com/radareorg/radare2/blob/ab7a4b804545a80f496ccf93228d1d14f86adbbb/libr/include/r_socket.h#L86-L87